### PR TITLE
[DOCS] Uniform formatting for ES|QL commands

### PR DESCRIPTION
**Pull request by abdonpijpelink**
Applies uniform formatting (Syntax/Parameters/Description/Examples) for the reference docs of the ES|QL commands.

Preview: https://elasticsearch_101728.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/esql-commands.html

### CODE REVIEW

File: docs/reference/esql/processing-commands/stats.asciidoc [hunk 1]
```diff
@@ -1,8 +1,48 @@
...
+omitted, the output table contains exactly one row with the aggregations applied
+over the entire dataset.
+
+The following aggregation functions are supported:
+
+include::../functions/aggregation-functions.asciidoc[tag=agg_list]
+
+NOTE: `STATS` without any groups is much much faster than adding a group.
+
+NOTE: Grouping on a single field is currently much more optimized than grouping
```
Comment (alex-spies) [inline docs/reference/esql/processing-commands/stats.asciidoc:37]: thought (out of scope of this PR): throughout the changes in this PR, we seem to prefer to use `column` (and seem to reserve `field` for actual documents' fields - which I think makes great sense). I am unsure whether this comment should apply to columns instead.

File: docs/reference/esql/processing-commands/stats.asciidoc [hunk 2]
```diff
@@ -1,8 +1,48 @@
...
+----
+
+*Parameters*
+
+`columnX`::
+The name by which the aggregated value is returned. If omitted, the name is
+equal to the corresponding expression (`expressionX`).
+
+`expressionX`::
+An expression that calculates an aggregated value.
```
Comment (alex-spies) [inline docs/reference/esql/processing-commands/stats.asciidoc:19]: bikeshed comment, but `compute` may be more standard in this context, no?
```suggestion
An expression that computes an aggregated value.
```
However, `calculate` may already be used quite frequently in the rest of the docs, so I'm not sure it's worth splitting hairs now.

File: docs/reference/esql/source-commands/from.asciidoc [hunk 3]
```diff
@@ -2,10 +2,32 @@
...
-The `FROM` source command returns a table with up to 10,000 documents from a
-data stream, index, or alias. Each row in the resulting table represents a
-document. Each column corresponds to a field, and can be accessed by the name
-of that field.
+**Syntax**
+
+[source,esql]
+----
+FROM index_pattern [METADATA fields]
+----
```
Comment (alex-spies) [inline docs/reference/esql/source-commands/from.asciidoc:10]: praise: having the grammar spelled out is great.

File: docs/reference/esql/source-commands/from.asciidoc [hunk 4]
```diff
@@ -2,10 +2,32 @@
...
+
+*Description*
+
+The `FROM` source command returns a table with data from a data stream, index,
+or alias. Each row in the resulting table represents a document. Each column
+corresponds to a field, and can be accessed by the name of that field.
+
+By default, `FROM` retrieves up to 500 rows. Use <<esql-limit>> to set a another
+limit. You can increase the number of rows up to 10,000. `FROM` does not return
+more than 10,000 rows, regardless of the `LIMIT` command's value.
```
Comment (alex-spies) [inline docs/reference/esql/source-commands/from.asciidoc:28]: praise: very helpful update of the docs here!

File: docs/reference/esql/processing-commands/limit.asciidoc [hunk 5]
```diff
@@ -2,12 +2,28 @@
...
+
+`max_number_of_rows`::
+The maximum number of rows to return.
+
+*Description*
+
+The `LIMIT` processing command enables you to limit the number of rows that are
+returned. If not specified, `LIMIT` defaults to `500`.
+
+A query does not return more than 10,000 rows, regardless of the `LIMIT` value.
```
Comment (alex-spies) [inline docs/reference/esql/processing-commands/limit.asciidoc:22]: out of scope: It may be worth noting that there are settings that can be overridden to adjust these values; in fact, Nhat and I just made the settings dynamic, so one can update them with a `PUT /_cluster/settings`.

**Review by alex-spies — APPROVED** [review]
LGTM!

File: docs/reference/esql/processing-commands/limit.asciidoc (hunk 5 above)
Comment (abdonpijpelink) [reply docs/reference/esql/processing-commands/limit.asciidoc:22]: Oh, that's good to know! Thanks. I'll add that to the Limitations docs.

File: docs/reference/esql/processing-commands/stats.asciidoc (hunk 1 above)
Comment (abdonpijpelink) [reply docs/reference/esql/processing-commands/stats.asciidoc:37]: Yeah, I've been trying to use "column" consistently, but every now and then a "field" slips through. Thanks!

File: docs/reference/esql/processing-commands/limit.asciidoc (hunk 5 above)
Comment (alex-spies) [reply docs/reference/esql/processing-commands/limit.asciidoc:22]: The settings are called `esql.query.result_truncation_max_size` and `esql.query.result_truncation_default_size`.

I'm currently trying to backport a PR that correctly enables setting them on 8.11 as well.

File: docs/reference/esql/processing-commands/sort.asciidoc [hunk 6]
```diff
@@ -1,35 +1,56 @@
...
+----
+SORT column1 [ASC/DESC][NULLS FIRST/NULLS LAST][, ..., columnN [ASC/DESC][NULLS FIRST/NULLS LAST]]
+----
+
+*Parameters*
+
+`columnX`::
+The column to sort on.
+
+*Description*
```
Comment (astefan) [inline docs/reference/esql/processing-commands/sort.asciidoc:17]: Since the original version of the documentation for `sort`, there has been an [improvement](https://github.com/elastic/elasticsearch/pull/98337) made to this command: If a column that is being sorted on has multi-values, then if the ordering is `asc` we will use the `min` value from a multi-value row. If the ordering is `desc` we will use the `max` value from a multi-value row.

Comment (astefan) [reply docs/reference/esql/processing-commands/sort.asciidoc:17]: Similar to what ES does but with only two non-modifiable options: https://www.elastic.co/guide/en/elasticsearch/reference/current/sort-search-results.html#_sort_mode_option

Comment (abdonpijpelink) [reply docs/reference/esql/processing-commands/sort.asciidoc:17]: Thanks! I've added the following sentence: _Sorting on multivalued columns uses the lowest value when sorting ascending and the highest value when sorting descending._

File: docs/reference/esql/processing-commands/stats.asciidoc [hunk 7]
```diff
@@ -1,8 +1,49 @@
...
+The following aggregation functions are supported:
+
+include::../functions/aggregation-functions.asciidoc[tag=agg_list]
+
+NOTE: `STATS` without any groups is much much faster than adding a group.
+
+NOTE: Grouping on a single column is currently much more optimized than grouping
+      on many column. In some tests we have seen grouping on a single `keyword`
+      column to be five times faster than grouping on two `keyword` columns. Do 
+      not try to work around this combining the two columns together with
```
Comment (astefan) [inline docs/reference/esql/processing-commands/stats.asciidoc:40]:
```suggestion
      not try to work around this by combining the two columns together with 
```

File: docs/reference/esql/source-commands/from.asciidoc [hunk 8]
```diff
@@ -2,10 +2,34 @@
...
+`fields`::
+A comma-separated list of <<esql-metadata-fields,metadata fields>> to retrieve.
+
+*Description*
+
+The `FROM` source command returns a table with data from a data stream, index,
+or alias. Each row in the resulting table represents a document. Each column
+corresponds to a field, and can be accessed by the name of that field.
+
+By default, `FROM` retrieves up to 500 rows. Use <<esql-limit>> to set a another
```
Comment (astefan) [inline docs/reference/esql/source-commands/from.asciidoc:26]: I have somewhat mixed feelings about this section. `FROM` in itself doesn't limit the number of rows, it's the `limit` command that does this. Using a query like `FROM test` will have by default a `limit 500` added to it, but this is slightly different than saying that `FROM retrieves up to 500 rows`. Imho, this section should be removed.

Comment (astefan) [reply docs/reference/esql/source-commands/from.asciidoc:26]: I've checked the `limit` docs section and I believe it's spelled out clearly there what we do with these default numbers.
Maybe reword this one for `FROM` to something like: 
> By default, a query without an explicit <<esql-limit>> will return up to 500 rows. This also applies to `FROM` command, by piping a default `LIMIT 500` to a query like `FROM test`. Using <<esql-limit>> you can increase the number of rows up to 10,000. ES|QL queries will not return more than 10,000 rows, regardless of the `LIMIT` command's value. You can change the maximum number of rows with the `esql.query.result_truncation_max_size` static cluster setting."

Comment (abdonpijpelink) [reply docs/reference/esql/source-commands/from.asciidoc:26]: That's a good point. The implicit limit has nothing to do with `FROM`. It is worth calling out here though, as I think it's something users will run into when they write their first ES|QL query in Discover. How about we turn it into the following "NOTE" callout instead:

image

We can then leave the 10,000 max discussion to the LIMIT docs.

Comment (astefan) [reply docs/reference/esql/source-commands/from.asciidoc:26]: Perfect 👍

**Review by astefan — APPROVED** [review]
LGTM
